src/main/java/org/sagebionetworks/web/client/jsinterop/react/HasStyle.java

```diff
@@ -0,0 +1,54 @@
+package org.sagebionetworks.web.client.jsinterop.react;
+
+import jsinterop.base.JsPropertyMap;
+import org.sagebionetworks.web.client.jsinterop.JsObject;
+import org.sagebionetworks.web.client.jsinterop.PropsWithStyle;
+import org.sagebionetworks.web.client.widget.ReactComponent;
+
+/**
+ * Abstract class for React component widgets that have a style prop. The style prop for the component may be manipulated
+ * to show/hide the component based on the current state of the widget.
+ * @param <T> the prop type.
+ */
+public abstract class HasStyle<T extends PropsWithStyle>
```

Comment:
One of the trickier parts of having React components in widgets is manipulating the visibility. When we inject ReactComponent widgets as children of other ReactComponents, the child components do not have the root element where we would typically add `display: none`.

We can get around this by manipulating the props of the children. For children with a `style` prop, we can append `display: none` to `style`.